lesson4/hometask_frameq.rb

```diff
@@ -0,0 +1,29 @@
+class LazyHash < Hash
+  def method_missing(name)
+    key = name.to_s
+    super.method_missing(key) unless self.key?(key)
+    if self[key].class == Proc
+      self[key] = self[key].call
+      self[key]
+    end
+    self[key]
+  end
+
+  def lazy?(key)
+    self[key.to_s].class == Proc ? true : false
```

Comment:
The same as above.
And **never** wrap such simple condition to return explicitly `true` or `false`. It's redundant.
You line `self[key.to_s].class == Proc` anyway will return  `true` or `false` why you try to wrap it?